.github/workflows/main.yml

```diff
@@ -46,11 +94,20 @@ jobs:
           fetch-depth: 0
       - uses: actions/setup-node@8c91899e586c5b171469028077307d293428b516 # tag=v3
         with:
+          cache: npm
           node-version: lts/*
-          cache: 'npm'
       - run: npm ci
-      # Branches that will release new versions are defined in .releaserc.json
+        # Branches that will release new versions are defined in .releaserc.json
       - run: npx semantic-release
+        # Don't allow interrupting the release step if the job is cancelled, as it can lead to an inconsistent state
+        # e.g. git tags were pushed but it exited before `npm publish`
+        if: always()
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_PUBLISH_TOKEN }}
+        # Re-run semantic release with rich logs if it failed to publish for easier debugging
+      - run: npx semantic-release --dry-run --debug
```

Comment:
The `--dry-run` here is intentional, as it's not always safe to re-run a botched release. Worst case scenario you could end up with two broken releases on `npm` instead of just the one 😅 The `--debug` is still really useful even if used with `--dry-run` and gives you additional context that the `npx semantic-release` step won't have in its log output,

Reply:
This is the main.yml for pkg-utils and not the template I think?

Reply:
There used to be just one `npx semantic-release` command in the main workflow. This PR introduces an additional one that only runs if the first one fails. It's conceptually like a retry function with `--debug` so you can get more information on why it failed, and the `--dry-run` is an extra safeguard. Sometimes a package can be misconfigured such that it will generate git tags, bump the version and generate a change log but fail to publish the npm package due to 2FA set to its strictest setting (where it doesn't allow automation tokens). 
In this scenario the `--dry-run` ensures broken git tags is only pushed once even though semantic release runs twice.